### PR TITLE
feat: drop if exists

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This closes https://github.com/GreptimeTeam/greptimedb/issues/2794.

@zuston it happens that I have some time to review this issue, if you're working on this issue, feel free to take any value from my PR and prioritize your patch.

@sunng87 I noticed that we actually check table existence before submit DropTable DDL task. If so, we can simply filter drop table tasks in `StatemenExecutor::drop_table` and assert table exists for the following producer.

Also, I'd like to confirm whether DDL task procedures can run concurrently. That is, if two `DROP TABLE t` can run concurrently with `t` exists. If so, we may need some lock/sync when removing table metadata.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by tisonkun** [comment]
Maybe I should add some e2e tests. But let's first take a look at the impl direction.

File: src/common/meta/src/rpc/ddl.rs [hunk 1]
```diff
@@ -176,6 +179,7 @@ pub struct DropTableTask {
     pub schema: String,
     pub table: String,
     pub table_id: TableId,
+    pub drop_if_exists: bool,
```
Comment (WenyXu) [inline src/common/meta/src/rpc/ddl.rs:182]: For backward compatibility.
```suggestion
    #[serde(default)]
    pub drop_if_exists: bool,
```

**Review by WenyXu — APPROVED** [review]
Rest LGTM, Let's add some sqlness tests.

File: src/common/meta/src/rpc/ddl.rs (hunk 1 above)
Comment (tisonkun) [reply src/common/meta/src/rpc/ddl.rs:183]: Updated.

**Comment by tisonkun** [comment]
Updated. PTAL again.

**Review by MichaelScofield — APPROVED** [review]

**Comment by MichaelScofield** [comment]
Drop table procedures cannot run concurrently because they acquire the same procedure lock key. The key locks on the table name.

**Comment by tisonkun** [comment]
Is this an unstable case - https://github.com/GreptimeTeam/greptimedb/actions/runs/7095139148/job/19311581914?

Since we pass at the first attempt.

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**